src/wallet/wallet.cpp

```diff
@@ -3290,7 +3294,7 @@ bool CWallet::AttachChain(const std::shared_ptr<CWallet>& walletInstance, interf
             }
         }
         walletInstance->m_attaching_chain = false;
-        walletInstance->chainStateFlushed(ChainstateRole::NORMAL, chain.getTipLocator());
+        walletInstance->chainStateFlushed(ChainstateRole::NORMAL);
```

Comment:
You could probably trigger the behavior by adding a sleep call here. When a reorg is processed in-between the  `ScanForWalletTransactions` call and this `chainStateFlushed` call.